crates/starknet-types-rpc/src/block_id.rs

```diff
@@ -0,0 +1,67 @@
+use serde::{Deserialize, Deserializer, Serialize};
```

Comment:
Why are the types in this file manually created and not generated?

Reply:
`BlockId` is used a lot and its actual representation is:

```rust
enum BlockId {
    Tag(BlockId),
    Hash(BlockIdHash),
    Number(BlockIdNumber),
}

struct BlockIdHash {
    block_hash: BlockHash,
}

struct BlockIdNumber {
    block_number: BlockNumber, // u64
}
```

which is not ideal because it's used a lot and it's a hassle to construct.

For this reason, I created a custom implementation of `Serialize` and `Deserialize` to make it just

```rust
enum BlockId {
    Tag(BlockTag),
    Hash(BlockHash),
    Number(BlockNumber),
}
```

For `SyncingStatus`, it's roughly the same.

It's "normal" representation is:

```rust
enum SyncingStatus {
    False(bool), // must always be `false`
    Syncing(SyncState),
}
```

Which I replaced by:

```rust
enum SyncingState {
    NotSyncing, // serializes as `false`, deserializes only `false
    Syncing(SyncState),
}
```

This is mainly for convenience and to reduce boilerplate in downstream crates.

Reply:
Not that I know of, but feel free to tell if you see any of them!